lib/messages.js

```diff
@@ -103,6 +103,49 @@ exports.connectRelay = {
   }
 }
 
+exports.mutableReq = {
+  preencode (state, m) {
+    cenc.uint.preencode(state, m.seq)
+    cenc.buffer.preencode(state, m.salt)
+  },
+  encode (state, m) {
+    cenc.uint.encode(state, m.seq)
+    cenc.buffer.encode(state, m.salt)
+  },
+  decode (state) {
+    return {
+      seq: cenc.uint.decode(state),
+      salt: cenc.buffer.decode(state)
+    }
+  }
+}
+
+exports.mutableRes = {
+  preencode (state, m) {
+    cenc.uint.preencode(state, m.seq)
+    cenc.buffer.preencode(state, m.value)
+    cenc.buffer.preencode(state, m.signature)
+    cenc.fixed32.preencode(state, m.publicKey)
+    cenc.buffer.preencode(state, m.salt)
+  },
+  encode (state, m) {
+    cenc.uint.encode(state, m.seq)
+    cenc.buffer.encode(state, m.value)
+    cenc.buffer.encode(state, m.signature)
+    cenc.fixed32.encode(state, m.publicKey)
+    cenc.buffer.encode(state, m.salt)
+  },
+  decode (state) {
+    return {
+      seq: cenc.uint.decode(state),
+      value: cenc.buffer.decode(state),
+      signature: cenc.buffer.decode(state),
+      publicKey: cenc.fixed32.decode(state),
+      salt: cenc.buffer.decode(state)
```

Comment:
salt is provided by the user in the response so it should not be sent over in the response